src/cmd/crane.go

```diff
@@ -104,15 +104,111 @@ func newRegistryCommand() *cobra.Command {
 	return cmd
 }
 
+type registryLoginOptions struct {
+	interactive   bool
+	originalRunFn func(cmd *cobra.Command, args []string) error
+}
+
+func (o *registryLoginOptions) run(cmd *cobra.Command, args []string) error {
+	if o.interactive {
+		uname, err := cmd.Flags().GetString("username")
+		if err != nil {
+			return err
+		}
+		pass, err := cmd.Flags().GetString("password")
+		if err != nil {
+			return err
+		}
+		passStdin, err := cmd.Flags().GetBool("password-stdin")
+		if err != nil {
+			return err
+		}
+		//Check if a server was provided
+		if len(args) == 0 {
+			var server string
+			prompt := &survey.Input{
+				Message: lang.CmdToolsRegistryLoginPromptServer,
+				Help:    lang.CmdToolsRegistryLoginPromptServerHelp,
+			}
+			err = survey.AskOne(prompt, &server, survey.WithValidator(survey.Required))
+			if err != nil {
+				return err
+			}
+			args = []string{server}
+		}
+
+		if uname == "" {
+			prompt := &survey.Input{
+				Message: lang.CmdToolsRegistryLoginPromptUsername,
+				Help:    lang.CmdToolsRegistryLoginPromptUsernameHelp,
+			}
+			err = survey.AskOne(prompt, &uname, survey.WithValidator(survey.Required))
+			if err != nil {
+				return err
+			}
+			cmd.Flags().Set("username", uname)
```

Comment:
Linting flagged here for error check
```suggestion
			err := cmd.Flags().Set("username", uname)
			if err != nil {
			  return err
			}
```